pkg/controller/usercleanup/user_cleanup_controller.go

```diff
@@ -0,0 +1,166 @@
+package usercleanup
+
+import (
+	"context"
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	crtCfg "github.com/codeready-toolchain/host-operator/pkg/configuration"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+	"time"
+)
+
+var log = logf.Log.WithName("controller_usercleanup")
+
+type StatusUpdater func(userAcc *toolchainv1alpha1.UserSignup, message string) error
+
+// Add creates a new UserCleanup Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager, crtConfig *crtCfg.Config) error {
+	return add(mgr, newReconciler(mgr, crtConfig))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager, crtConfig *crtCfg.Config) reconcile.Reconciler {
+	return &ReconcileUserCleanup{
+		client:    mgr.GetClient(),
+		scheme:    mgr.GetScheme(),
+		crtConfig: crtConfig,
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("usercleanup-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to the secondary resource UserSignup and requeue the owner UserSignup
+	if err := c.Watch(
+		&source.Kind{Type: &toolchainv1alpha1.UserSignup{}},
+		&handler.EnqueueRequestForOwner{
+			IsController: true,
+			OwnerType:    &toolchainv1alpha1.UserSignup{},
+		}); err != nil {
```

Comment:
correct me if I'm wrong - the owner of the `UserSignup` cannot be a `UserSignup`
you should use `handler.EnqueueRequestForObject{}` instead

Reply:
In addition, you could use a predicate to filter `UserSignups` with the label `toolchain.dev.openshift.com/state` equal to either `not-ready` or `deactivated`

Reply:
Are `not-ready` and `deactivated` the only states we should be interested in?  What about `banned` ?

Reply:
I would avoid relying on the state label in this predicate. If we add more states or rename existing ones we can forget updating this. It's not critical but less error prone if our predicate is simple.

Reply:
In this case do we even need a predicate?

Reply:
That would work. I'm fine with both options (do not have a predicate or filter out state == approved)

Reply:
I've removed the predicate for now.